docs/contributor/draft/rate-limit-api.md

```diff
@@ -0,0 +1,122 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | in case of empty, it's applied for the whole namespace          | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a selector                   | no                            |
+| localRateLimit                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| localRateLimit.by                                | object      | container for criteria you can limit by (for the future)        | yes                           |
```

Comment:
Any reason to have that `by` field? 
Wouldn't `localRateLimit.requests` or `localRateLimit.headerPresence` be equivalent?

Reply:
would be, I put it here since it felt nice to have it readable like a sentence

Reply:
do you think dropping this and avoiding unnecessary nesting would be worth more than this?

Reply:
Yes, this would be more similar to the strategy we took for API Gateway handler (ext auth, jwt, noauth)